bean-utils-library/src/main/java/com/hotels/beans/transformer/TransformerImpl.java

```diff
@@ -371,13 +388,56 @@ private String evalBreadcrumb(final String fieldName, final String breadcrumb) {
     }
 
     /**
-     * Retrieves the transformer function.
+     * Gets the source field type.
+     * @param sourceObjectClass the source object class
+     * @param sourceFieldName sourceFieldName the field name in the source object (if different from the target one)
+     * @return the source field type
+     */
+    private Class<?> getSourceFieldType(final Class<?> sourceObjectClass, final String sourceFieldName) {
+        String cacheKey = "SourceFieldType-" + sourceObjectClass.getName() + "-" + sourceFieldName;
+        return cacheManager.getFromCache(cacheKey, Class.class)
+                .orElseGet(() -> {
+                    Class<?> classType = null;
+                    try {
+                        classType = reflectionUtils.getDeclaredFieldType(sourceFieldName, sourceObjectClass);
+                    } catch (MissingFieldException e) {
+                        // in case the source field is a primitive type and the destination one is composite,
+                        // the source field type is returned without going in deep
+                        if (classUtils.isPrimitiveType(sourceObjectClass)) {
+                            classType = sourceObjectClass;
+                        } else if (!settings.isSetDefaultValueForMissingField()) {
+                            throw e;
+                        }
+                    }
+                    cacheManager.cacheObject(cacheKey, classType);
+                    return classType;
+                });
+    }
+
+    /**
+     * Retrieves the transformer functions. If the field type is different and they are primitive
+     * a conversion function is automatically added.
+     * @param sourceObjectClass the source object class
+     * @param sourceFieldName the source field name
      * @param field The field on which the transformation should be applied.
+     * @param isDestinationFieldPrimitiveType indicates if the destination field type is primitive or not
      * @param breadcrumb The full field path on which the transformation should be applied.
      * @return the transformer function.
      */
-    private FieldTransformer getTransformerFunction(final Field field, final String breadcrumb) {
-        return settings.getFieldsTransformers().get(settings.isFlatFieldNameTransformation() ? field.getName() : breadcrumb);
+    private List<FieldTransformer> getTransformerFunction(final Class<?> sourceObjectClass, final String sourceFieldName, final Field field,
```

Comment:
I think that the returned value of this method can be cached, this would speed up the performance.

Reply:
@AndreaMars94 I've applied this change.